Assets/_Functionalities/FirstPersonViewer/Scripts/FirstPersonViewerInput.cs

```diff
@@ -80,44 +94,67 @@ private void Update()
         {
             HandleCursorLocking();
 
+            isEditingInputfield = IsInputfieldSelected();
+
             HandleExiting();
         }
 
         private void HandleCursorLocking()
         {
-            //When editing an inputfield just block this function.
-            if (ShouldSkipCursorLocking()) return;
-
-            //When key is released release/lock mouse
-            if (ExitInput.WasReleasedThisFrame() && !isEditingInputfield)
+            if (!lockMouseModus)
             {
-                bool isLocked = Cursor.lockState == CursorLockMode.Locked;
-                ToggleCursor(isLocked);
+                //Use the normal locking mode (Mouse Click)
+                bool cursorLocked = isLocked;
+                if (LeftClick.triggered && !cursorLocked)
+                {
+                    if (isEditingInputfield) return;
+                    if (Interface.PointerIsOverUI()) return;
+
+                    ToggleCursor(false);
+                }
+                else if (LeftClick.WasReleasedThisFrame() && cursorLocked) ToggleCursor(true);
```

Comment:
kunnen al deze conditions hierboven en onder misschien even iets duidelijker worden opgeschreven? het is niet heel leesbaar